Index: sid/configure.in
===================================================================
--- sid.orig/configure.in
+++ sid/configure.in
@@ -1521,9 +1521,7 @@
   am/Makefile
   benchmark-suite/Makefile
   doc/Makefile
-  doc/goops/Makefile
   doc/r5rs/Makefile
-  doc/ref/Makefile
   doc/tutorial/Makefile
   emacs/Makefile
   examples/Makefile
Index: sid/doc/Makefile.am
===================================================================
--- sid.orig/doc/Makefile.am
+++ sid/doc/Makefile.am
@@ -21,21 +21,14 @@
 
 AUTOMAKE_OPTIONS = gnu
 
-SUBDIRS = ref tutorial goops r5rs
+SUBDIRS = tutorial r5rs
 
 dist_man1_MANS = guile.1
 
-EXAMPLE_SMOB_FILES = \
-    ChangeLog-2008 Makefile README image-type.c image-type.h myguile.c
-
 OLDFMT = oldfmt.c
 
 dist-hook:
 	cp $(srcdir)/$(OLDFMT) $(distdir)/
-	mkdir $(distdir)/example-smob
-	for f in $(EXAMPLE_SMOB_FILES); do \
-	  cp $(srcdir)/example-smob/$$f $(distdir)/example-smob/; \
-        done
 
 EXTRA_DIST = groupings.alist ChangeLog-2008 # guile-api.alist
 
